1.2/percentile.go

```diff
@@ -0,0 +1,27 @@
+package percentile
+
+import (
+	"errors"
+	"math"
+)
+
+type percentile func([]uint) (uint, error)
+
+func generatePercentile(x float32) percentile {
+
+	f := func(data []uint) (uint, error) {
+		if x > 1.0 || x < 0.0 {
+			return 0, errors.New("input out of range")
+		}
```

Comment:
This error should be returned by generatePercentile (which provides x).  By the time a caller calls `f` they're not in a position to do anything about x.  Whereas when they call generatePercentile, they're in a position to act on & provide a different x.

In other words keep `type percentile func([]uint) uint` and have `func generatePercentile(x float32) (percentile, error)`.